experiments/tribler_idle_run/tribler_idle_run.py

```diff
@@ -1,49 +1,56 @@
 #!/usr/bin/env python3
 import os
+import signal
 from asyncio import ensure_future, get_event_loop, sleep
 from pathlib import Path
 
+from tribler_core.components.base import Session
 from tribler_core.config.tribler_config import TriblerConfig
-from tribler_core.session import Session
+from tribler_core.start_core import components_gen
 
 BASE_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..', '..'))
 
 
-class IdleTribleRunner:
+async def idle_tribler_run():
+    state_dir = Path(os.path.abspath(os.path.join(BASE_DIR, "output", "tribler-state")))
+    config = TriblerConfig(state_dir=state_dir)
+    config.api.http_enabled = False
+    config.ipv8.port = 21000
+    config.libtorrent.port = 21005
 
-    def __init__(self):
-        self.session = None
+    components = components_gen(config)
+    session = Session(config, components)
+    signal.signal(signal.SIGTERM, lambda signum, stack: session.shutdown_event.set)
+    session.set_as_default()
 
-    async def run(self):
-        state_dir = Path(os.path.abspath(os.path.join(BASE_DIR, "output", "tribler-state")))
-        config = TriblerConfig(state_dir=state_dir)
-        config.api.http_enabled = False
-        config.ipv8.port = 21000
-        config.libtorrent.port = 21005
+    try:
+        await session.start_components()
+    except Exception as e:  # pylint: disable=broad-except
+        print(str(e))
+        return
 
-        self.session = Session(config)
-        try:
-            await self.session.start()
-        except Exception as e:
-            print(str(e))
-            get_event_loop().stop()
-        else:
-            print("Tribler started")
+    print("Tribler started")
 
-        if "TRIBLER_EXECUTION_TIME" in os.environ:
-            run_time = int(os.environ["TRIBLER_EXECUTION_TIME"])
-        else:
-            run_time = 60 * 10  # Run for 10 minutes by default
+    if "TRIBLER_EXECUTION_TIME" in os.environ:
+        run_time = int(os.environ["TRIBLER_EXECUTION_TIME"])
+    else:
+        run_time = 60 * 10  # Run for 10 minutes by default
 
-        await sleep(run_time)
+    await sleep(run_time)
 
-        print("Stopping Tribler idle run")
-        await self.session.shutdown()
-        get_event_loop().stop()
+    print("Stopping Tribler idle run")
+    session.shutdown_event.set()
+
+    # Indicates we are shutting down core. With this environment variable set
+    # to 'TRUE', RESTManager will no longer accept any new requests.
+    os.environ['TRIBLER_SHUTTING_DOWN'] = "TRUE"
+
+    await session.shutdown()
+
+    get_event_loop().stop()
```

Comment:
**Suggestion:** Maybe we should move this call to `stop()` outside this method? Also, is this necessary (we are already calling `close()`)?

Reply:
Thanks for pointing me this; it was an old code. Initially, I fixed the code of `tribler_idle_run` to make it run without errors, but I did not change the overall structure. Now I rewrote `tribler_idle_run` to streamline the code.